src/lib/util/Config.ts

```diff
@@ -264,26 +266,35 @@ export class Config {
 	}
 
 	private async initializeConfig(): Promise<void> {
+		// TODO: This method's current functionality is because we need to support v2 and v3 at the same time.
+		//  Once we no longer need to support v2, we can change this method to, for example, not use a separate config file
+		//  for v3.
 		this.logger.trace(`Initializing Config`);
+		// If there's no v2-based config, we need to create one, so the customer can safely downgrade to v2 if they want.
+		// NOTE: When v3 becomes the primary version, we may want to delete this, and possibly rework the config files
+		// in general.
+		if (!await this.fileHandler.exists(this.legacyConfigFilePath)) {
+			this.logger.trace(`No v2 config file exists. Creating one from defaults.`);
+			await this.createNewConfigFile(DEFAULT_CONFIG, this.legacyConfigFilePath);
+		}
+		// If there's no v3-based config, create one by directly copying the v2 config.
 		if (!await this.fileHandler.exists(this.configFilePath)) {
-			await this.createNewConfigFile(DEFAULT_CONFIG);
+			this.logger.trace(`No v3 config file exists. Creating one by copying v2`);
+			const v2ContentString = await this.fileHandler.readFile(this.legacyConfigFilePath);
+			const v2Content = JSON.parse(v2ContentString) as ConfigContent;
+			await this.createNewConfigFile(v2Content, this.configFilePath);
 		}
+
+		// Read the v3 config file and use that as our config.
 		const fileContent = await this.fileHandler.readFile(this.configFilePath);
 		this.logger.trace(`Config content to be set as ${fileContent}`);
 		this.configContent = JSON.parse(fileContent) as ConfigContent;
-
-		// TODO remove this logic before GA, as it is only necessary for short term migrations from old format.
```

Comment:
My reasoning for why it's okay to delete this is as follows:
- If they're installing v3 fresh, then they're inherently getting the new config format, so there's no concern there.
- If they're upgrading from v2, then they've presumably already run at least one v2 command, and so this line of code has already been run.
- It seems like the code was only necessary for our internal development anyway (though that's somewhat unclear)

Reply:
I believe this was needed when we moved from 1.x to 2.x. Seems alright to remove it now.